diff --git a/cocos/physics3d/CCPhysics3DConstraint.cpp b/cocos/physics3d/CCPhysics3DConstraint.cpp
index 7af4324..0532123 100644
--- a/cocos/physics3d/CCPhysics3DConstraint.cpp
+++ b/cocos/physics3d/CCPhysics3DConstraint.cpp
@@ -313,7 +313,7 @@ bool Physics3DHingeConstraint::getEnableAngularMotor() const
 }
 float Physics3DHingeConstraint::getMotorTargetVelosity() const
 {
-    return static_cast<btHingeConstraint*>(_constraint)->getMotorTargetVelosity();
+    return static_cast<btHingeConstraint*>(_constraint)->getMotorTargetVelocity();
 }
 float Physics3DHingeConstraint::getMaxMotorImpulse() const
 {
