diff --git a/cocos/2d/CCFontAtlas.cpp b/cocos/2d/CCFontAtlas.cpp
index 7253947e54..4f8d1f612c 100644
--- a/cocos/2d/CCFontAtlas.cpp
+++ b/cocos/2d/CCFontAtlas.cpp
@@ -130,7 +130,7 @@ FontAtlas::~FontAtlas()
 #if CC_TARGET_PLATFORM != CC_PLATFORM_WIN32 && CC_TARGET_PLATFORM != CC_PLATFORM_WINRT && CC_TARGET_PLATFORM != CC_PLATFORM_ANDROID
     if (_iconv)
     {
-        iconv_close(_iconv);
+        iconv_close((iconv_t)_iconv);
         _iconv = nullptr;
     }
 #endif
@@ -240,7 +240,7 @@ void FontAtlas::conversionU32TOGB2312(const std::u32string& u32Text, std::unorde
             size_t inLen = strLen * 2;
             size_t outLen = gb2312StrSize;
 
-            iconv(_iconv, (char**)&pin, &inLen, &pout, &outLen);
+            iconv((iconv_t)_iconv, (char**)&pin, &inLen, &pout, &outLen);
         }
 #endif
     }
