diff --git a/cocos/physics/CCPhysicsWorld.cpp b/cocos/physics/CCPhysicsWorld.cpp
index cd6effd..d27df13 100644
--- a/cocos/physics/CCPhysicsWorld.cpp
+++ b/cocos/physics/CCPhysicsWorld.cpp
@@ -488,12 +488,7 @@ bool PhysicsWorld::init()
 {
     do
     {
-#if CC_TARGET_PLATFORM == CC_PLATFORM_WINRT || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
 		_cpSpace = cpSpaceNew();
-#else
-        _cpSpace = cpHastySpaceNew();
-        cpHastySpaceSetThreads(_cpSpace, 0);
-#endif
         CC_BREAK_IF(_cpSpace == nullptr);
         
         cpSpaceSetGravity(_cpSpace, PhysicsHelper::point2cpv(_gravity));
@@ -903,11 +898,7 @@ void PhysicsWorld::update(float delta, bool userCall/* = false*/)
 
     if (userCall)
     {
-#if CC_TARGET_PLATFORM == CC_PLATFORM_WINRT || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
 		cpSpaceStep(_cpSpace, delta);
-#else
-		cpHastySpaceStep(_cpSpace, delta);
-#endif
     }
     else
     {
@@ -919,11 +910,7 @@ void PhysicsWorld::update(float delta, bool userCall/* = false*/)
             while(_updateTime>step)
             {
                 _updateTime-=step;
-#if CC_TARGET_PLATFORM == CC_PLATFORM_WINRT || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
 				cpSpaceStep(_cpSpace, dt);
-#else
-				cpHastySpaceStep(_cpSpace, dt);
-#endif
 			}
         }
         else
@@ -933,11 +920,7 @@ void PhysicsWorld::update(float delta, bool userCall/* = false*/)
                 const float dt = _updateTime * _speed / _substeps;
                 for (int i = 0; i < _substeps; ++i)
                 {
-#if CC_TARGET_PLATFORM == CC_PLATFORM_WINRT || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
 					cpSpaceStep(_cpSpace, dt);
-#else
-					cpHastySpaceStep(_cpSpace, dt);
-#endif 
 					for (auto& body : _bodies)
                     {
                         body->update(dt);
@@ -1000,11 +983,7 @@ PhysicsWorld::~PhysicsWorld()
     removeAllBodies();
     if (_cpSpace)
     {
-#if CC_TARGET_PLATFORM == CC_PLATFORM_WINRT || CC_TARGET_PLATFORM == CC_PLATFORM_WIN32
 		cpSpaceFree(_cpSpace);
-#else
-		cpHastySpaceFree(_cpSpace);
-#endif 
     }
     CC_SAFE_RELEASE_NULL(_debugDraw);
 }
