diff --git a/cocos/editor-support/spine/SkeletonRenderer.cpp b/cocos/editor-support/spine/SkeletonRenderer.cpp
index b8668e7335..051924a279 100644
--- a/cocos/editor-support/spine/SkeletonRenderer.cpp
+++ b/cocos/editor-support/spine/SkeletonRenderer.cpp
@@ -636,7 +636,7 @@ AttachmentVertices* SkeletonRenderer::getAttachmentVertices (spMeshAttachment* a
 	return (AttachmentVertices*)attachment->rendererObject;
 }
 
-Rect SkeletonRenderer::getBoundingBox () const {
+cocos2d::Rect SkeletonRenderer::getBoundingBox () const {
 	float minX = FLT_MAX, minY = FLT_MAX, maxX = -FLT_MAX, maxY = -FLT_MAX;
 	float scaleX = getScaleX(), scaleY = getScaleY();
 	for (int i = 0; i < _skeleton->slotsCount; ++i) {
@@ -663,7 +663,7 @@ Rect SkeletonRenderer::getBoundingBox () const {
 	}
 	Vec2 position = getPosition();
     if (minX == FLT_MAX) minX = minY = maxX = maxY = 0;    
-	return Rect(position.x + minX, position.y + minY, maxX - minX, maxY - minY);
+	return cocos2d::Rect(position.x + minX, position.y + minY, maxX - minX, maxY - minY);
 }
 
 // --- Convenience methods for Skeleton_* functions.
diff --git a/cocos/platform/CCGLView.h b/cocos/platform/CCGLView.h
index 36f2fdf57b..f1a6d98ef9 100644
--- a/cocos/platform/CCGLView.h
+++ b/cocos/platform/CCGLView.h
@@ -37,7 +37,7 @@ THE SOFTWARE.
 #endif /* (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) */
 
 #if (CC_TARGET_PLATFORM == CC_PLATFORM_MAC)
-typedef void* id;
+typedef struct objc_object *id;
 #endif /* (CC_TARGET_PLATFORM == CC_PLATFORM_MAC) */
 
 #if (CC_TARGET_PLATFORM == CC_PLATFORM_WIN32) || (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX)
